source/material_model/visco_plastic.cc

```diff
@@ -39,6 +39,11 @@ namespace aspect
       for ( unsigned int i=0; i < x_comp.size(); ++i)
         x_comp[i] = std::min(std::max(x_comp[i], 0.0), 1.0);
 
+      // assign compositional fields associated with strain a value of 0
+      if (use_strain_weakening == true)
```

Comment:
This will work, but it is not the preferred way to go (what if another plugin for heating or a postprocessor also requires the first fields to be something special?). If you feel like going the 'correct' way, you can require the integrated strain fields to be always named "strain_xx" and so on, and set here the compositions with the indices `this->introspection().compositional_index_for_name("strain_xx")` to zero. This way you can also check if there actually are these fields, or if somebody just switched on 'use_strain_weakening' and forgot to add the fields.